src/client/pfs.go

```diff
@@ -438,6 +442,114 @@ func (c APIClient) ListBlock() ([]*pfs.BlockInfo, error) {
 	return blockInfos.BlockInfo, nil
 }
 
+// PutObject puts a value into the object store and tags it with 0 or more tags.
+func (c APIClient) PutObject(r io.Reader, tags ...string) (object *pfs.Object, retErr error) {
+	w, err := c.newPutObjectWriteCloser(tags...)
+	if err != nil {
+		return nil, sanitizeErr(err)
+	}
+	defer func() {
+		if err := w.Close(); err != nil && retErr == nil {
+			retErr = sanitizeErr(err)
+		}
+		if retErr == nil {
+			object = w.object
+		}
+	}()
+	if _, err := io.CopyBuffer(w, r, make([]byte, MaxMsgSize/2)); err != nil {
+		return nil, sanitizeErr(err)
+	}
+	// return value set by deferred function
+	return nil, nil
+}
+
+// GetObject gets an object out of the object store by hash.
+func (c APIClient) GetObject(hash string, writer io.Writer) error {
+	getObjectClient, err := c.ObjectAPIClient.GetObject(
+		c.ctx(),
+		&pfs.Object{Hash: hash},
+	)
+	if err != nil {
+		return sanitizeErr(err)
+	}
+	if err := grpcutil.WriteFromStreamingBytesClient(getObjectClient, writer); err != nil {
+		return sanitizeErr(err)
+	}
+	return nil
+}
+
+// ReadObject gets an object by hash and returns it directly as []byte.
+func (c APIClient) ReadObject(hash string) ([]byte, error) {
+	var buffer bytes.Buffer
+	if err := c.GetObject(hash, &buffer); err != nil {
+		return nil, err
+	}
+	return buffer.Bytes(), nil
+}
+
+// TagObject applies a tag to an existing object.
+func (c APIClient) TagObject(hash string, tags ...string) error {
```

Comment:
Hmm, not sure I follow, are you saying that the signature would be:

```
TagObject(tags ...string)
```

It seems like that would be a weird signature since you could call it in ways that didn't really make sense, ie with a single `tag`.

Implementation wise I still kind of like distinguishing `tags` and `hashes` from each other since they come from different places and I can imagine a number of weird bugs that could happen if we accidentally overwrote a hash with a tag.